scripts/_helpers.py

```diff
@@ -150,3 +148,89 @@ def dlProgress(count, blockSize, totalSize):
 
     urllib.request.urlretrieve(url, file, reporthook=dlProgress)
 
+
+def mocksnakemake(rulename, **wildcards):
+    """
+    This function is expected to be executed in the 'scripts'-directory within
+    the snakemake project. It returns a snakemake.script.Snakemake object,
+    based on the Snakefile of the upper directory, such that includes
+    mostly all necessary information (input, output, log etc.).
+
+    If a rule has wildcards, you have to specify them in **wildcards.
+
+    Parameters
+    ----------
+    rulename: str
+        name of the rule for which the snakemake object should be generated
+    **wildcards:
+        keyword arguments fixing the wildcards. Only necessary if wildcards are
+        needed.
+    """
+    import snakemake as sm
+    import os
+    import logging
+    from pypsa.descriptors import Dict
+    from os.path import abspath
+
+    old_wd = os.getcwd()
+    os.chdir(base_dir)
```

Comment:
@euronion the `base_dir` is not defined. A good suggestion for determining it (rather then setting it to '..')

Reply:
I think passing it as a function argument is reasonable:

```
    def mocksnakemake(rulename, base_dir="..", **wildcards):
```

Reply:
`base_dir` is now set to  `Path(__file__).parent.parent`. You agree with that?

Reply:
No, this should not work the way you want it to, as you cannot got beyond your anchor, in this case `__name__`, see [the pathlib doc](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.parent).

```
p = Path(__file__)  # "_helpers"
p.parent            # "."
p.parent.parent     # "."
```

You can either hard-code `base_dir = ".."`, which I think is os-portable and the easier to understand version.
Or you have to resolve the absolute path first:

```
p = Path(__file__) # "_helpers"
p = p.absolute()   # "/home/user/pypsa-eur/scripts/_helpers"
p.parent           # "/home/user/pypsa-eur/scripts"
p.parent.parent    # "/home/user/pypsa-eur"
```

Reply:
@FabianHofmann : The missing key information is that, if you execute a script as `python relative/path/test.py`, the `__file__` variable in `test.py` will only hold `relative/path/test.py` and not its absolute path, like it is if you `import test` or use `run` in ipython.

I think it makes sense to use `__file__` to locate the correct directory, but am split between
1. `Path(__file__).resolve().parent.parent`, and
2. `Path(__file__).parent / ".."`

Reply:
I am a bit confused now, on my system `resolve()` is insufficient:

```
In [1]: import pathlib

In [2]: p = pathlib.Path(__name__)

In [3]: p
Out[3]: WindowsPath('__main__')

In [4]: p.resolve()
Out[4]: WindowsPath('__main__')

In [5]: p.absolute()
Out[5]: WindowsPath('C:/Users/__main__')
```

If we want to determine the parent of the current working directory, the solution should be
```
pathlib.Path.cwd().parent
```

If we want to locate the great-parent directory of the scripts, then semantically I would suggest to go with @coroa 
```
Path(__file__).aboslute().parent.parent
```

Reply:
ah okay, thank you both for clarifying! I tested it now with to 

```
Path(__file__).parent.joinpath('..', 'logs', f"{snakemake.rule}.log")
```
which works well in for both `mocksnakemake` and normal `snakemake`

Reply:
@euronion It's about `__file__` not `__name__`, which is not defined in the interpreter only, well in a file. On an existing file `resolve()` should work.